apps/discovery_api/test/unit/discovery_api/event_handler_test.exs

```diff
@@ -65,4 +68,20 @@ defmodule DiscoveryApi.EventHandlerTest do
       assert_called(DataJsonService.delete_data_json())
     end
   end
+
+  describe "handle_event/1 #{dataset_delete()}" do
+    test "tells the data json plug to delete its current data json cache" do
+      allow(DiscoveryApi.RecommendationEngine.delete(any()), return: :seriously_delete)
+      allow(StatsCalculator.delete_completeness(any()), return: :whatever_seriously)
+      allow(ResponseCache.invalidate(), return: :seriously_do_not_care)
+      allow(SystemNameCache.delete(any(), any()), return: :its_very_serious)
+      allow(DataJsonService.delete_data_json(), return: :ok)
+
+      dataset = TDG.create_dataset(%{})
+
+      Brook.Event.process(:discovery_api, Brook.Event.new(type: dataset_delete(), data: dataset, author: :author))
+
+      assert_called(DataJsonService.delete_data_json())
```

Comment:
It would be nice to have some more assertions in here, as well as a test for the failure case (for the rescue you have) for this and then also an integration test.

- Ben and Kevin

Reply:
👍

Reply:
@JarredOlson Please share your thoughts